### PR TITLE
#197 Handle meetings with no attendees, add delete action

### PR DESCRIPTION
**Pull request by jaydonkrooss**
#197 Addresses the issue we've been seeing with meetings without attendees. My solution is to just handle the issue client-side, allowing users to view all affected meetings and manually remove them. 

Test plan to replicate local development environment:
1. On a manage meeting page, add a user to the meeting queue (for in-person meeting)
2. Click "ready for attendee" to admit the user to the "Meetings in Progress" section
3. In local DB, delete the attendee from the officehours_api_attendee table (where meeting_id is from the id from officehours_api_meeting table): `delete from officehours_api_attendee where meeting_id={meeting_id}`
4. On browser, open the console and refresh the page. You should see a meeting with the message "Invalid meeting ID:" and no attendee. It should have a trash icon to delete the meeting
5. Delete the meeting by pressing the delete button and confirming. Double check the database to see that the meeting info is fully cleared

To test the meeting without attendees in just the Meeting Queue, repeat the above but skip over step 2

### CODE REVIEW

**Comment by jxiao21** [comment]
Works as you described. Just making sure I understand, the goal is that if there is a meeting with no attendees, that it will say somewhere that the meeting is invalid and that it needs to be deleted? I think it may help to make it a little more clear to the user with something a little more eye-catching that the meeting should be deleted if that's the case.

**Comment by jaydonkrooss** [comment]
> Works as you described. Just making sure I understand, the goal is that if there is a meeting with no attendees, that it will say somewhere that the meeting is invalid and that it needs to be deleted? 

Yes, the invalid data should be presented as opposed to crashing the page, which we saw before.

> I think it may help to make it a little more clear to the user with something a little more eye-catching that the meeting should be deleted if that's the case.

Good suggestion, I'll look into some error message or at least changing the text inside the table to be more clear.

**Comment by jaydonkrooss** [comment]
@jxiao21 I just added a suggestion to delete the meeting in the table entries. This is a rare scenario to begin with so I think that will suffice to indicate to the user that the invalid meeting should be removed.

File: src/assets/src/components/meetingTables.tsx [hunk 1]
```diff
@@ -138,7 +145,22 @@ function UnstartedMeetingEditor (props: UnstartedMeetingEditorProps) {
 }
 
 function StartedMeetingEditor (props: MeetingEditorProps) {
-    const attendee = props.meeting.attendees[0];
+    const attendee = props.meeting.attendees[0] ?? null;
+    if (!attendee) return (
```
Comment (zqian) [inline src/assets/src/components/meetingTables.tsx:149]: line 148-163 is identical to line 84-98. Could reformat to use one single source.

**Review by zqian — APPROVED** [review]
Following the test plan and tested the feature works as expected. Left reformatting comments.

File: src/assets/src/components/meetingTables.tsx [hunk 2]
```diff
@@ -80,13 +80,20 @@ interface UnstartedMeetingEditorProps extends MeetingEditorProps, AssigneeSelect
...
 function UnstartedMeetingEditor (props: UnstartedMeetingEditorProps) {
-    const attendee = props.meeting.attendees[0];
+    const attendee = props.meeting.attendees[0] ?? null;
     if (!attendee) return (
         <>
         <td>No attendee</td>
         <td></td>
-        <td>Invalid meeting ID: {props.meeting.id}</td>
-        <td></td>
+        <td>Invalid meeting ID: {props.meeting.id}. Please remove this meeting</td>
```
[the remark this answers is not in the document]
Comment (jaydonkrooss) [reply src/assets/src/components/meetingTables.tsx]: added a period

File: src/assets/src/components/meetingTables.tsx (hunk 1 above)
Comment (jaydonkrooss) [reply src/assets/src/components/meetingTables.tsx]: I went ahead and made the "Invalid Meeting" component generic

**Pull request merged**